src/Extensions/TimedScopes/TaskExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Drawing.Drawing2D;
+using System.Threading.Tasks;
+
+namespace Microsoft.Omex.Extensions.TimedScopes
+{
+	/// <summary>
+	/// Extensions for Task to wrap them in TimedScope
+	/// </summary>
+	public static class TaskExtensions
+	{
+		/// <summary>
+		/// Starts TimedScopes before task execution and finish after it's completion
+		/// </summary>
+		public static async ValueTask<T> WithTimedScope<T>(this ValueTask<T> task, ITimedScopeProvider provider, string name)
+		{
+			using TimedScope timedScope = provider.Start(name, TimedScopeResult.SystemError);
+			T result = await task.ConfigureAwait(false);
```

Comment:
Just `await` within the return to allow the other operation to be performed on another thread. It's only a minor optimisation, but easy to implement.

Reply:
Sorry, I'm not sure that I understand what you suggesting. Could you please explain more?

Reply:
I've added comments to clarify that TimedScope results should be set after successful task completion